plugins/trace_api_plugin/include/eosio/trace_api/extract_util.hpp

```diff
@@ -39,6 +39,27 @@ inline transaction_trace_v0 to_transaction_trace_v0( const chain::transaction_tr
    return r;
 }
 
+inline transaction_trace_v1 to_transaction_trace_v1( const cache_trace& t ) {
+    transaction_trace_v1 r;
+    if( !t.trace->failed_dtrx_trace ) {
+        r.id = t.trace->id;
+    } else {
+        r.id = t.trace->failed_dtrx_trace->id; // report the failed trx id since that is the id known to user
+    }
+    r.status = t.trace->receipt->status;
+    r.cpu_usage_us = t.trace->receipt->cpu_usage_us;
+    r.net_usage_words = t.trace->receipt->net_usage_words;
```

Comment:
As is `to_transaction_trace_v1` can't be called with an empty `optional t.trace->receipt` however, with the addition of `status` I can see someone adding failed in the future. Add a check that `receipt` exists.